### PR TITLE
BNGP-4313: Calculate tasklist based on session values and journey definition

### PR DESCRIPTION
**Pull request by paulsimonandrews**

### CODE REVIEW

File: packages/webapp/src/journey-validation/task-list-generator.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+const getTaskList = (journey, session) => {
+  const applicantInfoTask = getJourneySectionStatus(journey, REGISTRATION.APPLICANT_INFO, session)
+  const landOwnershipTask = getJourneySectionStatus(journey, REGISTRATION.LAND_OWNERSHIP, session)
+  const siteBoundaryTask = getJourneySectionStatus(journey, REGISTRATION.SITE_BOUNDARY, session)
+  const habitatInfoTask = getJourneySectionStatus(journey, REGISTRATION.HABITAT_INFO, session)
+  const legalAgreementTask = getJourneySectionStatus(journey, REGISTRATION.LEGAL_AGREEMENT, session)
+  const localLandChargeTask = getJourneySectionStatus(journey, REGISTRATION.LOCAL_LAND_CHARGE, session)
+
```
Comment (KiranVarma02) [inline packages/webapp/src/journey-validation/task-list-generator.js:150]: To discuss - I am still thinking: 
```suggestion

```
```suggestion

```
```suggestion

```


```
const taskSectionFactory = (taskTitle, tasks, id) => {
  return {
    taskTitle,
    tasks: tasks.map(task => {
      return {
        title: task.title,
        status: task.status,
        url: task.url,
        id: task.id
      }
    })
  }
}

```

Comment (KiranVarma02) [inline packages/webapp/src/journey-validation/task-list-generator.js:150]: thought in progress -

```
const taskSectionFactory = (taskTitle, tasks, id) => {
  return {
    taskTitle,
    tasks: tasks.map(task => {
      return {
        title: task.title,
        status: task.status,
        url: task.url,
        id: task.id
      }
    })
  }
}

```

Comment (KiranVarma02) [inline packages/webapp/src/journey-validation/task-list-generator.js:150]: thought in progress -


```suggestion

const taskSectionFactory = (taskTitle, tasks, id) => {
  return {
    taskTitle,
    tasks: tasks.map(task => {
      return {
        title: task.title,
        status: task.status,
        url: task.url,
        id: task.id
      }
    })
  }
}
```

Comment (KiranVarma02) [reply packages/webapp/src/journey-validation/task-list-generator.js:150]: I don't mean to add it to at this line obviously :). I just put that here for us to discuss later

Comment (KiranVarma02) [reply packages/webapp/src/journey-validation/task-list-generator.js:150]: refactoring tickets to follow

**Review by KiranVarma02 — DISMISSED** [review]
If testers confirm all functionality is working then we can go with what we have. We agreed to refactor later, so I am happy to approve.

**Review by pwadmore-ea — APPROVED** [review]

**Review by pwadmore-ea — APPROVED** [review]

**Review by pwadmore-ea — APPROVED** [review]

**Comment by shaundefra** [comment]
tested locally

**Pull request merged**